internal/gatewayapi/globalresources.go

```diff
@@ -0,0 +1,63 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package gatewayapi
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/envoyproxy/gateway/internal/gatewayapi/resource"
+	"github.com/envoyproxy/gateway/internal/ir"
+	"github.com/envoyproxy/gateway/internal/utils"
+)
+
+const envoyTLSSecretName = "envoy"
+
+// ProcessGlobalResources processes global resources that are not tied to a specific listener or route
+func (t *Translator) ProcessGlobalResources(resources *resource.Resources, gateways []*GatewayContext, xdsIR resource.XdsIRMap) error {
+	// Get the envoy client TLS secret. It is used for envoy to establish a TLS connection with control plane components,
+	// including the rate limit server and the wasm HTTP server.
+	envoyTLSSecret := resources.GetSecret(t.ControllerNamespace, envoyTLSSecretName)
+	if envoyTLSSecret == nil {
+		return fmt.Errorf("envoy TLS secret %s/%s not found", t.ControllerNamespace, envoyTLSSecretName)
+	}
+
+	for _, gw := range gateways {
+		key := utils.NamespacedName(gw).String()
```

Comment:
you may want to iterate over xdsIR instead, so you dont skip Its related to merge gateways